src/providers/GaugeProvider.tsx

```diff
@@ -0,0 +1,81 @@
+import { BasicPool, BasicPoolsContext } from "./BasicPoolsProvider"
+import React, { ReactElement, useContext, useEffect, useState } from "react"
+import { useGaugeController, useHelperContract } from "../hooks/useContract"
+import { BigNumber } from "ethers"
+
+export type Gauge = {
+  address: string
+  lpToken: string
+  name: string
+  symbol?: string
+  gaugeWeight: BigNumber
+  poolAddress: string
+  gaugeRelativeWeight: BigNumber
+  workingSupply?: BigNumber
+}
+
+export type Gauges = {
+  gaugeCount: number
+  gauges: Gauge[]
+}
+
+const initialGaugesState: Gauges = {
+  gaugeCount: 0,
+  gauges: [],
+}
+
+export const GaugeContext = React.createContext<Gauges>(initialGaugesState)
+
+export default function GaugeProvider({
+  children,
+}: React.PropsWithChildren<unknown>): ReactElement {
+  const gaugeController = useGaugeController()
+  const helperContract = useHelperContract()
+  const poolsContext = useContext(BasicPoolsContext)
```

Comment:
Interestingly, I don't think we need to have pools a dependency of gauges. 

We can fetch `LiquidityGaugeV5.LP_TOKEN()` and use that as a key. Then whenever we need to combine pool and gauge data (eg in the pie chart, or on the pools page when displaying rewards) we can do 
```
pools.map(pool => {
  // .. do pool stuff
  if (gauges[pool.lpToken]) {
    // do gauge stuff
  }})
  ```

This is what I've been referring to when I said "the lpToken is the foreign key between pools and gauges"